packages/sdk/src/store/store.ts

```diff
@@ -257,32 +264,81 @@ export class Store implements IStore {
     };
   }
 
-  private async getPeerToUse(): Promise<PeerId | undefined> {
-    let peerId: PeerId | undefined;
+  private async getPeerToUse(pubsubTopic: string): Promise<PeerId | undefined> {
+    const peers = await this.filterConnectedPeers(pubsubTopic);
+
+    const peer = this.options.peers
+      ? await this.getPeerFromConfigurationOrFirst(peers, this.options.peers)
+      : peers[0]?.id;
+
+    return peer;
+  }
 
-    if (this.options?.peer) {
-      const connectedPeers = await this.connectionManager.getConnectedPeers();
+  private async getPeerFromConfigurationOrFirst(
```

Comment:
nit: long function name

Reply:
agree, do you have another better name for it? 
otherwise I would keep present one